src/QMCWaveFunctions/einspline_spo.hpp

```diff
@@ -150,15 +150,16 @@ struct einspline_spo
       pos_type end(1);
       einsplines.resize(nBlocks);
       RandomGenerator<T> myrandom(11);
-      Array<T, 3> data(nx, ny, nz);
-      std::fill(data.begin(), data.end(), T());
-      myrandom.generate_uniform(data.data(), data.size());
       for (int i = 0; i < nBlocks; ++i)
       {
         einsplines[i] = myAllocator.createMultiBspline(T(0), start, end, ng, PERIODIC, nSplinesPerBlock);
         if (init_random)
-          for (int j = 0; j < nSplinesPerBlock; ++j)
-            myAllocator.set(data.data(), einsplines[i], j);
+          for (int j = 0; j < nSplinesPerBlock; ++j) {
+            // Generate different coefficients for each orbital
+            Array<T, 3> coef_data(nx+3, ny+3, nz+3);
+            myrandom.generate_uniform(coef_data.data(), coef_data.size());
```

Comment:
It is not clear to me how fast to generating 1GB random numbers.
I think we can randomize the coef_data for each block but not for each orbital.

Reply:
I suspect generating random numbers is faster than running the solver, but I didn't measure it.
Regardless of performance issues there are tradeoffs:
1. Use the same coefficients for each orbital.  This allows an easy check that the evaluation of each orbital gets the same answer (if printing a list of psi values, for instance).  However, it could miss cases where the wrong orbital is being read.
2. Use different coefficients for each orbital.  Checks the reverse of (1).  Given that the check against the reference SPO should catch errors in evaluation, but could potential miss problems if all the orbitals are the same, this seems like the more powerful option.

Reply:
Running the solver is extremely slow. Generating coefs can be slow but I also didn't measure it.
Actually, the coef_data is allocated and deallocated over and over. Please move it outside nblock loop.

I agree with your concern 2.
How about only randomize the first and last orbital of each tile.

Reply:
I think we should go with all of the orbitals separate for now - it's simple and straightforward.  If performance becomes an issue, the scheme you suggest can be implemented.